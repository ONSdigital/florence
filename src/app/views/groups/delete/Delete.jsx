import React from "react";
import PropTypes from "prop-types";
import Loader from "../../../components/loader/Loader";

const Delete = ({ id, name, deleteGroup, openModal, closeModal, loading }) => {
    const popout = {
        id: "delete",
        title: "Are you sure you want to delete this preview team?",
        body: "Team members cannot view content linked to this preview team after it has been deleted.",
        buttons: [
            {
                onClick: () => handleSubmit(),
                text: "Delete preview team",
                style: "warning",
            },
            {
                onClick: () => closeModal(["delete"]),
                text: "Cancel",
                style: "invert-primary",
            },
        ],
    };

    const handleClick = e => {
        e.preventDefault();
        openModal(popout);
    };

    const handleSubmit = () => {
        closeModal(["delete"]);
        deleteGroup(id, name);
    };

    return (
        <div className="grid__col-11 grid__col-md-9">
            {loading ? (
                <Loader />
            ) : (
                <div role="alert" className="panel panel--error panel--no-title margin-top--1 margin-bottom--1">
                    <h2 className="panel__error font-size--16">
                        <strong>Delete preview team</strong>
                    </h2>
                    <p>Team members cannot view content linked to this preview team after it has been deleted.</p>
                    <button className="btn btn--warning margin-top--1 margin-bottom--1" onClick={handleClick}>
                        Delete team
                    </button>
                </div>
            )}
        </div>
    );
};

Delete.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    deleteGroup: PropTypes.func.isRequired,
    openModal: PropTypes.func.isRequired,
    closeModal: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
};

export default Delete;
