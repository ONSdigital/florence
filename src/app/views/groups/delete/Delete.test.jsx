import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/extend-expect";
import Delete from "./Delete";

const defaultProps = {
    id: "0",
    deleteGroup: jest.fn(),
    openModal: jest.fn(),
    closeModal: jest.fn(),
    loading: false,
};
const handleSubmit = jest.fn();

describe("Delete", () => {
    it("matches the snapshot", () => {
        const tree = renderer.create(<Delete {...defaultProps} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it("when deleting is in progress matches the snapshot", () => {
        const props = { ...defaultProps, isLoading: true };
        const tree = renderer.create(<Delete {...props} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it("requests to open Modal", () => {
        render(<Delete {...defaultProps} />);

        userEvent.click(screen.getByRole("button", { name: "Delete team" }));

        expect(defaultProps.openModal).toBeCalledWith(
            expect.objectContaining({
                body: "Team members cannot view content linked to this preview team after it has been deleted.",
                id: "delete",
                title: "Are you sure you want to delete this preview team?",
            })
        );
    });
});
