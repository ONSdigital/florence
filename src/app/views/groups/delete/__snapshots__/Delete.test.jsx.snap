// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete matches the snapshot 1`] = `
<div
  className="grid__col-11 grid__col-md-9"
>
  <div
    className="panel panel--error panel--no-title margin-top--1 margin-bottom--1"
    role="alert"
  >
    <h2
      className="panel__error font-size--16"
    >
      <strong>
        Delete preview team
      </strong>
    </h2>
    <p>
      Team members cannot view content linked to this preview team after it has been deleted.
    </p>
    <button
      className="btn btn--warning margin-top--1 margin-bottom--1"
      onClick={[Function]}
    >
      Delete team
    </button>
  </div>
</div>
`;

exports[`Delete when deleting is in progress matches the snapshot 1`] = `
<div
  className="grid__col-11 grid__col-md-9"
>
  <div
    className="panel panel--error panel--no-title margin-top--1 margin-bottom--1"
    role="alert"
  >
    <h2
      className="panel__error font-size--16"
    >
      <strong>
        Delete preview team
      </strong>
    </h2>
    <p>
      Team members cannot view content linked to this preview team after it has been deleted.
    </p>
    <button
      className="btn btn--warning margin-top--1 margin-bottom--1"
      onClick={[Function]}
    >
      Delete team
    </button>
  </div>
</div>
`;
